### PR TITLE
Multithreading

### PR DESCRIPTION
**Pull request by Muhib-ur-Rehman**

### CODE REVIEW

File: src/com/company/multiThreading/Main.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        DemoThread demoThread2 = new DemoThread();
+        Thread thread2 = new Thread(demoThread2);
+        thread2.start();
+        thread1.join();
+        thread2.join();
+
+        try {
+            for (int i =0; i <5 ; i++){
+                System.out.println(Thread.currentThread().getName()+" : "+i);
+                Thread.sleep(1000);
```
Comment (mansoor-naseer) [inline src/com/company/multiThreading/Main.java:18]: @Muhib-ur-Rehman You should **not** be using **sleep()**!
Please read task instruction carefully

File: src/com/company/multiThreading/Main.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        thread1.start();
+        DemoThread demoThread2 = new DemoThread();
+        Thread thread2 = new Thread(demoThread2);
+        thread2.start();
+        thread1.join();
+        thread2.join();
+
+        try {
+            for (int i =0; i <5 ; i++){
+                System.out.println(Thread.currentThread().getName()+" : "+i);
```
Comment (mansoor-naseer) [inline src/com/company/multiThreading/Main.java:17]: @Muhib-ur-Rehman Why printing threadname here?, it is already printed in demothread.

**Review by mansoor-naseer — CHANGES_REQUESTED** [review]

File: src/com/company/multiThreading/Main.java (hunk 1 above)
Comment (Muhib-ur-Rehman) [reply src/com/company/multiThreading/Main.java:18]: Oh sorry, I will correct this.

File: src/com/company/multiThreading/Main.java (hunk 2 above)
Comment (Muhib-ur-Rehman) [reply src/com/company/multiThreading/Main.java:17]: Here it is used to print the main thread name

**Pull request merged**